### ISSUE TITLE
yoda's exceptRange option has false positive

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#7676]
**Tell us about your environment**

* **ESLint Version:** 3.11.1
* **Node Version:** chrome 55 (our online demo)
* **npm Version:** -

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

- nothing

**What did you do? Please include the actual source code causing the issue.**

```js
/*eslint yoda: [error, never, {exceptRange: true}] */

function get(list, index) {
    if (0 <= index && index < list.length) {
        return list[index]
    }
}
```

**What did you expect to happen?**

No errors since the condition is a range check.

**What actually happened? Please include the actual, raw output from ESLint.**

[yoda](http://eslint.org/docs/rules/yoda) rule is warning the code.

![image](https://cloud.githubusercontent.com/assets/1937871/20735233/17c1d048-b6e2-11e6-929e-ca2380038996.png)
